megamek/src/megamek/client/ui/swing/BoardEditor.java

```diff
@@ -1330,18 +1330,21 @@ private void updateWhenSelected() {
     }
 
     public void boardNew() {
-        RandomMapDialog rmd = new RandomMapDialog(frame, this, null, mapSettings);
-        rmd.setVisible(true);
-        board = BoardUtilities.generateRandom(mapSettings);
-        game.setBoard(board);
-        curfile = null;
-        frame.setTitle(Messages.getString("BoardEditor.title")); //$NON-NLS-1$
-        menuBar.setBoard(true);
-        bvc.doLayout();
+    	RandomMapDialog rmd = new RandomMapDialog(frame, this, null, mapSettings);
+    	boolean userCancel = rmd.activateDialog();
+    	if (!userCancel) {
+    		board = BoardUtilities.generateRandom(mapSettings);
+    		game.setBoard(board);
+    		curfile = null;
+    		frame.setTitle(Messages.getString("BoardEditor.title")); //$NON-NLS-1$
+    		menuBar.setBoard(true);
+    		bvc.doLayout();
+    		resetUndo();
+    	}
     }
-    
+
     public void boardResize() {
-        ResizeMapDialog emd = new ResizeMapDialog(frame, this, null, mapSettings);
+    	ResizeMapDialog emd = new ResizeMapDialog(frame, this, null, mapSettings);
```

Comment:
Formatting on this line is weird.

Reply:
Didn't actively change that line. But I noticed that my formatter was still using tabs which Ive changed now. I'd rather leave this for now and replace tabs across some files at another time.